daemon/container_unix.go

```diff
@@ -636,12 +636,12 @@ func (container *Container) updateJoinInfo(ep libnetwork.Endpoint) error {
 func (container *Container) updateEndpointNetworkSettings(n libnetwork.Network, ep libnetwork.Endpoint) error {
 	networkSettings := &network.Settings{NetworkID: n.ID(), EndpointID: ep.ID()}
 
-	networkSettings, err := container.buildPortMapInfo(n, ep, networkSettings)
+	networkSettings, err := container.buildPortMapInfo(ep, networkSettings)
```

Comment:
Since the err will not be used in other place but just return if it was not `nil`, why not use 
```
if networkSettings, err := container.buildPortMapInfo(ep, networkSettings); err != nil {
    return err
}
```